File: tests/fixtures/go/basic-crypto/pkg/kdf/pbkdf2.go
package kdf

import (
	"crypto/sha256"

	"golang.org/x/crypto/pbkdf2"
)

const (
	DefaultIterations = 100000
	DefaultKeyLength  = 32
)

// DeriveKey derives a key using PBKDF2 with default parameters
func DeriveKey(password, salt []byte) []byte {
	return pbkdf2.Key(password, salt, DefaultIterations, DefaultKeyLength, sha256.New)
}

// DeriveKeyCustom derives a key with custom iterations
func DeriveKeyCustom(password, salt []byte, iterations int) []byte {
	return pbkdf2.Key(password, salt, iterations, 32, sha256.New)
}

// DeriveKeyLiteral uses literal values directly
func DeriveKeyLiteral(password, salt []byte) []byte {
	return pbkdf2.Key(password, salt, 10000, 32, sha256.New)
}

File: tests/fixtures/python/basic-crypto/kdf/pbkdf2.py
"""PBKDF2 key derivation functions."""

import hashlib

DEFAULT_ITERATIONS = 100000
DEFAULT_KEY_LENGTH = 32


def derive_key(password: bytes, salt: bytes) -> bytes:
    """Derive a key using PBKDF2 with default parameters."""
    return hashlib.pbkdf2_hmac(
        "sha256", password, salt, DEFAULT_ITERATIONS, DEFAULT_KEY_LENGTH
    )


def derive_key_custom(password: bytes, salt: bytes, iterations: int) -> bytes:
    """Derive a key with custom iterations."""
    return hashlib.pbkdf2_hmac("sha256", password, salt, iterations, 32)


def derive_key_literal(password: bytes, salt: bytes) -> bytes:
    """Derive a key with literal values."""
    return hashlib.pbkdf2_hmac("sha256", password, salt, 10000, 32)
